pkg/auth/api/password_validation.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2025 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+
+	"github.com/bucketeer-io/bucketeer/v2/pkg/auth"
+	"github.com/bucketeer-io/bucketeer/v2/pkg/locale"
+	authproto "github.com/bucketeer-io/bucketeer/v2/proto/auth"
+)
+
+func validateUpdatePasswordRequest(
```

Comment:
This should ensure that the password is different. Otherwise, it could fail due to no rows being changed in MySQL.